frame/utility/src/lib.rs

```diff
@@ -349,7 +365,8 @@ decl_module! {
 						m.approvals.insert(pos, who.clone());
 						<Multisigs<T>>::insert(&id, call_hash, m);
 						Self::deposit_event(RawEvent::MultisigApproval(who, timepoint, id, call_hash));
-						return Ok(())
+						// Call is not made, so we can return that weight
+						return Ok(Some(call.get_dispatch_info().weight).into())
```

Comment:
The API is that you return the `actual_weight` not the refund. So this is doing the opposite of what you want. You could factor out your weight calculation into a function that you call here and from the weight annotation to avoid duplication.

Reply:
fixed, good catch